test/test_utils/ec2.py

```diff
@@ -298,13 +298,24 @@ def launch_instance(
     return response["Instances"][0]
 
 
+def get_available_reservation(ec2_client, instance_type, min_availability=1):
+    reservations = ec2_client.describe_capacity_reservations()
+    for reservation in reservations["CapacityReservations"]:
+        if (
+            reservation["InstanceType"] == instance_type
+            and reservation["AvailableInstanceCount"] >= min_availability
+        ):
+            return reservation
```

Comment:
To capture offline conversation, iteration may exacerbate the race condition since subsequent iterations will have increased time delay between checking for instance availability and attempting to spin up the instance. It maybe better to simple fall back and allow the function to retry instead.